<a href="https://colab.research.google.com/github/EkaterinaBoRu/Seminar8/blob/main/%D0%94%D0%97%E2%84%966.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import scipy.stats as stats
import numpy as np

**Задача 2**. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95

In [2]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [4]:
x_1= np.mean(x)
x_1

6.590000000000001

In [6]:
D_1=np.var(x, ddof=1)
D_1

0.2032222222222223

In [8]:
t1=stats.t.ppf(0.975,9)
t1

2.2621571627409915

In [9]:
x_1-t1*np.sqrt(D_1/10)

6.267515851415713

In [10]:
x_1+t1*np.sqrt(D_1/10)

6.912484148584288

Ответ: (6.26; 6.91)



**Задача 3**.Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170 Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175 Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [15]:
a = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])

In [16]:
b = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

In [17]:
len(a)

10

In [18]:
len (b)

10

In [19]:
a_1= np.mean(a)
a_1

166.2

In [20]:
b_1= np.mean(b)
b_1

168.1

In [22]:
delta= a_1-b_1
delta

-1.9000000000000057

In [23]:
D1=np.var(a, ddof=1)
D1

91.06666666666666

In [24]:
D2=np.var(b, ddof=1)
D2

60.10000000000001

In [25]:
D= (D1+D2)/2
D

75.58333333333334

In [26]:
SE=np.sqrt(D/10+D/10)
SE

3.8880157750022915

In [27]:
t=stats.t.ppf(0.975,18)
t

2.10092204024096

In [28]:
L= delta- t*SE
L

-10.068418034506857

In [29]:
U=  delta + t*SE
U

6.268418034506846

Ответ: (-10.07; 6.27)